listenbrainz/webserver/views/settings.py

```diff
@@ -242,6 +233,34 @@ def refresh_service_token(service_name: str):
     return jsonify({"access_token": user["access_token"]})
 
 
+@settings_bp.route('/music-services/<service_name>/connect/', methods=['POST'])
+@api_login_required
+def music_services_connect(service_name: str):
+    """ Connect last.fm/libre.fm account to ListenBrainz user. """
+    # TODO: add support for libre.fm
+    if service_name.lower() != "lastfm":
+        raise APINotFound("Service %s is invalid." % (service_name,))
+
+    data = request.json
+    if "external_user_id" not in data:
+        raise APIBadRequest("Missing 'external_user_id' in request.")
+    if "latest_listened_at" not in data:
+        raise APIBadRequest("Missing 'latest_listened_at' in request.")
+
+    try:
+        latest_listened_at = datetime.fromisoformat(data["latest_listened_at"])
+    except (ValueError, TypeError):
+        raise APIBadRequest(f"Value of latest_listened_at '{data['latest_listened_at']} is invalid.")
```

Comment:
Making `latest_listened_at` optional
```suggestion
    latest_listened_at = None
    if data["latest_listened_at"] is not None:
        try:
            latest_listened_at = datetime.fromisoformat(data["latest_listened_at"])
        except (ValueError, TypeError):
            raise APIBadRequest(f"Value of latest_listened_at '{data['latest_listened_at']} is invalid.")
```